src/Microsoft.Health.SqlServer/ManagedIdentitySqlConnectionBuilder.cs

```diff
@@ -14,25 +14,23 @@ namespace Microsoft.Health.SqlServer;
 public class ManagedIdentitySqlConnectionBuilder : ISqlConnectionBuilder
 {
     private readonly ISqlConnectionStringProvider _sqlConnectionStringProvider;
-    private readonly IAccessTokenHandler _accessTokenHandler;
     private readonly SqlRetryLogicBaseProvider _sqlRetryLogicBaseProvider;
-    private readonly string _azureResource = "https://database.windows.net/";
+    private readonly IAzureTokenCredentialProvider _azureTokenCredentialProvider;
 
     public ManagedIdentitySqlConnectionBuilder(
+        IAzureTokenCredentialProvider azureTokenCredentialProvider,
```

Comment:
Do we need yet another abstraction, or can we simply rely on `AzureCredential` directly? Then in the registration method, we pass in the appropriate credential based on the type.

Reply:
I added it to support unit testing of ManagedIdentitySqlConnection.

Reply:
But can't this connection builder just use `AzureCredential` instead of this abstraction? Like, we could just call this thing like `AzureCredentialSqlConnectionBuilder`